Makefile

```
@@ -1,56 +1,48 @@
-RESET_COLOR=\033[0m
-GREEN_COLOR=\033[32;01m
 
-CONTAINER_NAME := 'splunk'
+CONTAINER_NAME := "splunk"
 
 .PHONY: docs
 docs:
-	@echo "$(GREEN_COLOR)==> docs $(RESET_COLOR)"
-	@rm -rf ./docs/_build
-	@tox -e docs
-	@cd ./docs/_build/html && zip -r ../docs_html.zip . -x ".*" -x "__MACOSX"
-	@echo "$(ATTN_COLOR)==> Docs pages can be found at ./docs/_build/html, docs bundle available at ./docs/_build/docs_html.zip"
+	@make -C ./docs html
 
 .PHONY: test
 test:
-	@echo "$(GREEN_COLOR)==> test $(RESET_COLOR)"
-	@tox -e py -- ./tests
+	@python -m pytest ./tests
 
 .PHONY: test-unit
-test:
-	@echo "$(GREEN_COLOR)==> test $(RESET_COLOR)"
-	@tox -e py -- ./tests/unit
+test-unit:
+	@python -m pytest ./tests/unit
 
 .PHONY: test-integration
-test:
-	@echo "$(GREEN_COLOR)==> test $(RESET_COLOR)"
-	@tox -e py -- ./tests/integration ./tests/system
+test-integration:
+	@python -m pytest ./tests/integration ./tests/system
 
-.PHONY: up
-up:
-	@echo "$(GREEN_COLOR)==> up $(RESET_COLOR)"
+.PHONY: docker-up
+docker-up:
 	@docker-compose up -d
 
-.PHONY: remove
-remove:
-	@echo "$(GREEN_COLOR)==> rm $(RESET_COLOR)"
-	@docker-compose rm -f -s
-
-.PHONY: wait_up
-wait_up:
-	@echo "$(GREEN_COLOR)==> wait_up $(RESET_COLOR)"
-	@for i in `seq 0 180`; do if docker exec -it $(CONTAINER_NAME) /sbin/checkstate.sh &> /dev/null; then break; fi; printf "\rWaiting for Splunk for %s seconds..." $$i; sleep 1; done
-
-.PHONY: down
-down:
-	@echo "$(GREEN_COLOR)==> down $(RESET_COLOR)"
+.PHONY: docker-ensure-up
+docker-ensure-up:
+	@for i in `seq 0 180`; do \
+		if docker exec -it $(CONTAINER_NAME) /sbin/checkstate.sh &> /dev/null; then \
```

This command is really bad and i have been frustrated with it (while using podman), since it does not differentiate between errors returned by `docker`/`shell` and `checkstate.sh` errors.

```suggestion
		if docker exec -it $(CONTAINER_NAME) /bin/bash -c "/sbin/checkstate.sh &> /dev/null"; then \
```

This way you at least get an error that docker is not in `$PATH` and users will act accordingly (edit `MakeFile` to use podman or just do not use `docker-ensure-up`, instead of endless waiting).


```
[mateusz@MPOLIWCZ-M-3NJ6 splunk-sdk-python ]$ make docker-ensure-up
/bin/sh: docker: command not found
Waiting for Splunk for 0 seconds...^[[A/bin/sh: docker: command not found
Waiting for Splunk for 1 seconds.../bin/sh: docker: command not found
Waiting for Splunk for 2 seconds.../bin/sh: docker: command not found
Waiting for Splunk for 3 seconds.../bin/sh: docker: command not found
Waiting for Splunk for 4 seconds.../bin/sh: docker: command not found
```

Still not ideal, but at least the error is visible.